src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -170,19 +176,38 @@ public BufferedImage getUnitImage(AnimState state, int imageIndex)
   public void drawUnit(Graphics g, Unit u, AnimState state, int imageIndex, int drawX, int drawY)
   {
     boolean flipImage = SpriteMapView.shouldFlip(u);
+    boolean drawBuff = AnimState.IDLE == state && !u.CO.getActiveAbilityName().isEmpty();
 
     BufferedImage frame = getUnitImage(state, imageIndex);
+    BufferedImage buffFrame = drawBuff ? buffMask.getFrame(imageIndex) : null;
     int shiftX =(SpriteLibrary.baseSpriteSize - frame.getWidth())/2; // center X
     int shiftY = SpriteLibrary.baseSpriteSize - frame.getHeight(); // bottom-justify Y
 
     // Draw the unit, facing the appropriate direction.
     if( flipImage && isStateFlippable(state) )
     {
       g.drawImage(frame, drawX - shiftX + (frame.getWidth()), drawY + shiftY, -frame.getWidth(), frame.getHeight(), null);
+      if( drawBuff )
+      {
+        // Set opacity as a function of time.
+        double opacity = 0.3*Math.sin(System.currentTimeMillis()/250.) + 0.6;
+        ((Graphics2D)g).setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, (float)opacity));
+        g.drawImage(buffFrame, drawX - shiftX + (buffFrame.getWidth()),
+            drawY + shiftY, -buffFrame.getWidth(), buffFrame.getHeight(), null);
+        ((Graphics2D)g).setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 1.0f));
```

Comment:
What is this casting witchery?

Reply:
[This guy](https://stackoverflow.com/questions/19344878/difference-between-graphics-and-graphics2d#19344937) and [this guy](https://jvm-gaming.org/t/graphics-vs-graphics2d/38668/2) summarize it well.

tl:dr; Spooky witchcraft.

Reply:
That's... not really spooky witchcraft.
Do you really *need* to cast it, though?

Reply:
Yes. `Graphics` doesn't have the composite interface, but `Graphics2D` does. It just so happens that the only other subclass of `Graphics` is `DebugGraphics`, which is used by precisely nobody ever, so we can just cast to the newer `Graphics2D` object and use the cool new stuff.

It just inherits from the old interface for backwards compatibility.

Reply:
I'd lobby perhaps for moving to use `Graphics2D` explicitly in the method signatures, rather than just casting junk and assuming it works.
...then again, I imagine we'd probably still end up casting it at the top level?

Reply:
Yes. `getGraphics` doesn't hand out `Graphics2D` objects.

We could still do the cast at the top level and just have it that way everywhere, but that is definitely beyond the scope of this PR.